nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -574,6 +582,27 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
     " milliseconds to smooth path." << std::endl;
 #endif
 
+  if (_debug_visualizations) {
+    // plot footprint path planned for debug
+    if (_planned_footprints_smoothed_publisher->get_subscription_count() > 0) {
+      auto marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
+
+      visualization_msgs::msg::Marker clear_all_marker;
+      clear_all_marker.action = visualization_msgs::msg::Marker::DELETEALL;
+      marker_array->markers.push_back(clear_all_marker);
+
+      _planned_footprints_smoothed_publisher->publish(std::move(marker_array));
```

Comment:
This doesn't need to be a standalone message, it can just be the first marker in the markerarray I believe.

Reply:
Not sure I get it, the msg has to be a visualization_msgs::msg::MarkerArray, can't be a visualization_msgs::msg::Marker (because that's what the publisher is initialized with)